File: src/EducationSystem/EducationSystem.WebApp/App/src/models/Material.ts
import Model from './Model'
import DocumentFile from './DocumentFile'

export default class Material extends Model {
  public Name: string = ''
  public Template: string = ''
  public Files: Array<DocumentFile> = []
  public Anchors: Array<IMaterialAnchor> = []
}

export interface IMaterialAnchor {
  Id: number
  Name: string
  Token: string
}

File: src/EducationSystem/EducationSystem.Models/Rest/Material.cs
using System.Collections.Generic;
using EducationSystem.Models.Files;
using EducationSystem.Models.Rest.Basics;

namespace EducationSystem.Models.Rest
{
    public class Material : Model
    {
        public string Name { get; set; }

        public string Template { get; set; }

        public int? OwnerId { get; set; }

        public List<Document> Files { get; set; }

        public List<MaterialAnchor> Anchors { get; set; }

        public Material Format()
        {
            Name = Name?.Trim();
            Template = Template?.Trim();

            Anchors?.ForEach(x =>
            {
                x.Name = x.Name?.Trim();
                x.Token = x.Token?.Trim();
            });

            return this;
        }
    }
}